cli/azd/internal/repository/app_init.go

```diff
@@ -429,7 +449,103 @@ func prjConfigFromDetect(
 			name = config.Name
 		}
 		name = names.LabelName(name)
+		svc.Name = name
 		config.Services[name] = &svc
+
+		svcMapping[prj.Path] = name
+	}
+
+	if addResources {
+		config.Resources = map[string]*project.ResourceConfig{}
+		dbNames := map[appdetect.DatabaseDep]string{}
+
+		for database := range detect.Databases {
+			if database == appdetect.DbRedis {
+				redis := project.ResourceConfig{
+					Type: project.ResourceTypeDbRedis,
+					Name: "redis",
+				}
+				config.Resources[redis.Name] = &redis
+				dbNames[database] = redis.Name
+				continue
+			}
+
+			var dbType project.ResourceType
+			switch database {
+			case appdetect.DbMongo:
+				dbType = project.ResourceTypeDbMongo
+			case appdetect.DbPostgres:
+				dbType = project.ResourceTypeDbPostgres
+			}
+
+			db := project.ResourceConfig{
+				Type: dbType,
+			}
+
+			for {
+				dbName, err := promptDbName(i.console, ctx, database)
+				if err != nil {
+					return config, err
+				}
+
+				if dbName == "" {
+					i.console.Message(ctx, "Database name is required.")
+					continue
+				}
```

Comment:
nit. Consider a `Cancel` option here.  Or if `promptDbName` already supports user-cancellation with an error, please add a note about cancelling is handled with an error.

Reply:
User cancellation is supported via Ctrl+C as the standard prompt interfacing; I don't think we will want to introduce additional cancellation options here today.